scripts/cita.sh

```diff
@@ -348,108 +345,117 @@ node_up_check() {
 
-        append)
-            config "$@"
-            exit 0
-            ;;
-
-        setup)
-            do_setup
-            ;;
-
-        start)
-            # TODO: should not do so, but present tests need this
-            do_stop
-            node_down_check
-
-            # Make sure the RabbitMQ fresh
-            clear_rabbit_mq
-
-            do_start "$3" "$4"
-            ;;
-
-        stop)
-            node_up_check
-            do_stop
-            ;;
-
-        restart)
-            node_up_check
-            do_stop
-
-            # Make sure the RabbitMQ fresh
-            clear_rabbit_mq
-
-            do_start "$3" "$4"
-            ;;
-
-        ping)
-            do_ping
-            if [ "${PING_STATUS}" == "pong" ]; then
-                echo "pong"
-            else
-                echo "Node '${NODE_NAME}' not responding to pings."
-                exit 1
-            fi
-            ;;
-
-        top)
-            node_up_check
-            do_top
-            ;;
-        # deprecated, use 'top' instead
-        stat)
-            node_up_check
-            do_top
-            ;;
-        # similar to 'top', but ... ?
-        status)
-            do_status
-            ;;
-
-        logrotate)
-            do_logrotate
-            ;;
-
-        logs)
-            do_logs "$3"
-            ;;
-
-        backup)
-            node_down_check
-            do_backup
-            ;;
-
-        clean)
-            node_down_check
-            do_clean
-            ;;
-
-        *)
-            usage
-            ;;
+    append)
+        config "$@"
+        exit 0
+        ;;
+
+    setup)
+        do_setup
+        ;;
+
+    start)
+        # TODO: should not do so, but present tests need this
+        do_stop
+        node_down_check
+
+        # Make sure the RabbitMQ fresh
+        clear_rabbit_mq
+
+        do_start "$3" "$4"
+        ;;
+
+    stop)
+        node_up_check
+        do_stop
+        ;;
+
+    restart)
+        node_up_check
+        do_stop
+
+        # Make sure the RabbitMQ fresh
+        clear_rabbit_mq
+
+        do_start "$3" "$4"
+        ;;
+
+    ping)
+        do_ping
+        if [ "${PING_STATUS}" == "pong" ]; then
+            echo "pong"
+        else
+            echo "Node '${NODE_NAME}' not responding to pings."
+            exit 1
+        fi
+        ;;
+
+    top)
+        node_up_check
+        do_top
+        ;;
+    # deprecated, use 'top' instead
+    stat)
+        node_up_check
+        do_top
+        ;;
+    # similar to 'top', but ... ?
+    status)
+        do_status
+        ;;
+
+    logrotate)
+        do_logrotate
+        ;;
+
+    logs)
+        do_logs "$3"
+        ;;
+
+    backup)
+        node_down_check
+        do_backup
+        ;;
+
+    clean)
+        node_down_check
+        do_clean
+        ;;
+
+    *)
+        usage
+        ;;
 
     esac
 }
 
+# Test CITA is running in docker.
+check_in_docker() {
+    if grep docker /proc/1/cgroup -qa; then
```

Comment:
May i ask can it really check **cita process** in docker?

Reply:
It depends, but  I think it can cover more OS and docker version. More detail here:

https://stackoverflow.com/questions/23513045/how-to-check-if-a-process-is-running-inside-docker-container

But, If it don't work in some systems, you also can use `bebop` instead.